api/v1/_submissions/PKPBackendSubmissionsHandler.inc.php

```diff
@@ -67,109 +76,129 @@ public function authorize($request, &$args, $roleAssignments)
     }
 
     /**
-     * Get a list of submissions according to passed query parameters
+     * Get a collection of submissions
      *
      * @param $slimRequest Request Slim request object
      * @param $response Response object
+     * @param array $args arguments
      *
      * @return Response
      */
     public function getMany($slimRequest, $response, $args)
     {
-        $request = $this->getRequest();
+        $request = Application::get()->getRequest();
         $currentUser = $request->getUser();
         $context = $request->getContext();
 
-        // Merge query params over default params
-        $defaultParams = [
-            'count' => 20,
-            'offset' => 0,
-        ];
+        if (!$context) {
+            return $response->withStatus(404)->withJsonError('api.404.resourceNotFound');
+        }
+
+        $collector = $this->getSubmissionCollector($slimRequest->getQueryParams());
+
+        HookRegistry::call('API::submissions::params', [$collector, $slimRequest]);
 
-        // Anyone not a manager or site admin can only access their assigned
-        // submissions
-        $userRoles = $this->getAuthorizedContextObject(ASSOC_TYPE_USER_ROLES);
+        // Prevent users from viewing submissions they're not assigned to,
+        // except for journal managers and admins.
+        $userRoles = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_USER_ROLES);
         $canAccessUnassignedSubmission = !empty(array_intersect([ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER], $userRoles));
```

Comment:
At `if statement` on line 105, you're using a `!$canAccessUnassignedSubmission`. Being a double negation of the statement.

This couldn't be a `$cannotAccessUnassignedSubmission = array_intersect([ROLE_ID_SITE_ADMIN, ROLE_ID_MANAGER], $userRoles) === []` and at `if` you could use `$cannotAccessUnassignedSubmission &&`?

Reply:
This is a grammar preference, but I prefer boolean variables to be phrased in the affirmative. So `isIncomplete` instead of `notComplete`. This aligns the English-language description with the boolean value. So `true` is always a "yes/can/is" and `false` is always a `no/can't/isn't`.